src/_pytest/terminal.py

```diff
@@ -1007,9 +1007,15 @@ def _folded_skips(skipped):
     return values
 
 
+def _make_plural(count, key):
+    if key in ["error", "warning"] and count != 1:
+        key += "s"
+    return count, key
+
+
 def build_summary_stats_line(stats):
     known_types = (
-        "failed passed skipped deselected xfailed xpassed warnings error".split()
+        "failed passed skipped deselected xfailed xpassed warning error".split()
```

Comment:
That's an unnecessary API breakage, isn't it?  I.e. plugins might use `stats.get("warnings")`.
I think we should handle/set both then still, but can use "warning" only internally then.